### PR TITLE
Use the Terms UI to manage BP Types

### PR DESCRIPTION
**Pull request by imath**
Hi!

I know this wasn't the direction that was chosen for the BP Types UI, but I believe since this [comment](https://buddypress.trac.wordpress.org/ticket/7181#comment:10) was made, things have evolved.

1. As BuddyPress 7.0.0 will/could require WordPress 4.9 : term meta can now be used to store specific BP Types data.
2. We can keep on using a global registry for BP Types and use the WordPress Terms UI
3. Using a Post type to do something else than writing/publishing content seems weird to me. Moreover when we're talking about an object "Term" that should organize posts :)

I strongly believe the best way to maintain the BP Types UI in the long run is to use something that was primarly designed to manage terms > WP Terms UI.

In this PR, I'm adapting this WP Terms UI to match BP Types need:
- Ajax actions and bulk actions are neutralized,
- Quick edit row action is neutralized,
- Add/edit Forms and WP List Terms Table Columns are customized, 
- Only the BP Types needed data are editable.

Here's a screenshot of the BP Member Types UI:

member-types-manage

Here's a screenshot of the BP Group Types UI:

group-types-manage

**NB**

- BP Types registered by code cannot be deleted.
- Most BP Types registered by code data can be overridden by edits saved into the database (only slug and name cannot be edited).

It works fine on WordPress 5.4 + BuddyPress 7.0-alpha when WordPress is configured as a single site. Just like it was the case for the Post type scenario, some adjustments are needed when BuddyPress is network activated as `edit.php`, `post.php`, `edit-tags.php` and `edit-tag-form.php` are not available into the network admin.

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+		get_taxonomies( array( 'public' => false ) ),
+		array( bp_get_member_type_tax_name(), 'bp_group_type' )
+	);
+
+	$objects = array(
+		bp_get_member_type_tax_name() => 'user',
+		'bp_group_type'               => 'bp_group',
+	);
+
+	foreach ( $bp_type_taxonomies as $bp_type_taxonomy ) {
```
Comment (boonebgorges) [inline includes/functions.php]: Is this necessary just to enable 'show_ui'?

**Review by boonebgorges — COMMENTED** [review]
Thanks for working on this, @imath. I agree that this approach makes more sense than using a CPT, or the CPT UI in the Dashboard.

For navigation in the Network Admin, I think that submenu items under Groups and Users, which point to the relevant edit-tags.php page on the root blog, is probably fine. Not ideal - especially in the case of Groups, where you wouldn't have a top-level menu item to highlight as "current" on the root blog Dashboard - but maybe good enough for our purposes here.

File: includes/functions.php (hunk 1 above)
Comment (imath) [reply includes/functions.php]: Actually: no 😄 . We don't need to unregister/reregister taxonomies. Your question made me look deeper into WordPress and the `register_taxonomy_args` should be used to extend the BP Type taxonomy arguments. That's what I've done in https://github.com/buddypress/bp-types-ui/pull/6/commits/798f2af6960730297ada75918261182e43c3388f

Comment (boonebgorges) [reply includes/functions.php]: Nice!

**Comment by boonebgorges** [comment]
This looks pretty good to me, @imath - feels like it can be merged, except for the fact that there's no way to get to the screen if you're on Multisite. What do you think about that?

**Comment by imath** [comment]
Thanks for your feedback @boonebgorges. I think we could try to use a trick for multisite configs where BuddyPress is network activated:
- do as you suggested into network admin adding sub-menus for BP Types which point to the relevant edit-tags page of the root blog admin.
- into the root blog admin add a bp-groups menu which point to the network Groups screen and use it as the parent for the Group Types screen

I'll test it on a multisite config to see how it goes 😉

**Comment by imath** [comment]
@boonebgorges in https://github.com/buddypress/bp-types-ui/pull/6/commits/f014b29ab6f6990799e5702eb38b0d3ec16f62f2 I'm applying what I've described in my previous comment. This makes sure the admin can access to the BP Type Manage screens from the Network admin.

**Comment by boonebgorges** [comment]
Thank you!

**Pull request merged**